File: js/examples/rpc.ts
import { Address } from '@solana/addresses';
import { createSolanaRpc } from '@solana/rpc';
import { getStakeActivation } from 'solana-rpc-get-stake-activation';

const rpc = createSolanaRpc('https://api.testnet.solana.com');
let stake = '25R5p1Qoe4BWW4ru7MQSNxxAzdiPN7zAunpCuF8q5iTz';
let status = await getStakeActivation(rpc, stake as Address);

File: rust/examples/rpc.rs
use {
    solana_rpc_client::rpc_client::RpcClient,
    solana_rpc_get_stake_activation::get_stake_activation, solana_sdk::pubkey::Pubkey,
    std::str::FromStr,
};

fn main() {
    let stake = Pubkey::from_str("Bv33KFZT81nhk5FimTK1kNtz6fGh8fzzHSfLpbmDSsA5").unwrap();
    let rpc_client = RpcClient::new("https://api.testnet.solana.com");
    let activation_status = get_stake_activation(&rpc_client, &stake).unwrap();
    println!("Hello, {activation_status:?}!");
}
